### ISSUE TITLE
Document "types" module in README

### ISSUE THREAD

**Issue by NeQuissimus** [fthomas/refined#354]
Just stumbled upon `eu.timepit.refined.types.all`.
Let's add something about this package to the README :)

**Comment by fthomas** [comment]
Agreed! Most of the README was written before the `types` package was added.
In the early days of the library, the focus was on the predicates while it currently shifts more towards fully refined types. Nowadays I'm using type aliases like `PosInt` more than explicit types like `Int Refined Positive`.
